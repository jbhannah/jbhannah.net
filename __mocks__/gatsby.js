const gatsby = jest.requireActual("gatsby")

const data = {
  site: {
    siteMetadata: {
      siteUrl: "https://www.example.com",
      title: "Test Site",
      description: "Test site",
      twitterCreator: "@jbhannah",
      socialLinks: [
        {
          key: "Test Service",
          service: "Test Service",
          link: "https://www.example.com",
          name: "Test Name",
        },
      ],
      name: {
        first: "Jesse",
        mi: "B",
        last: "Hannah",
      },
      gender: "female",
      pronouns: ["she", "her"],
    },
  },
  file: {
    childImageSharp: {
      gatsbyImageData: {
        images: {
          fallback: {
            src: "avatar.png",
          },
          sources: [
            {
              type: "image/webp",
              srcSet: "avatar-srcSet.webp",
            },
          ],
        },
      },
    },
  },
}

module.exports = {
  ...gatsby,
  graphql: jest.fn(),
  useStaticQuery: jest.fn(() => data),
}
