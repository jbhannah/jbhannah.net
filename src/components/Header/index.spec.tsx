import { render, screen } from "@testing-library/react"
import * as Gatsby from "gatsby"
import * as React from "react"
import { Header } from "."

const data = {
  site: {
    siteMetadata: {
      socialLinks: [
        {
          key: "Test Service",
          service: "Test Service",
          link: "https://www.example.com",
          name: "Test Name",
        },
      ],
    },
  },
  file: {
    childImageSharp: {
      gatsbyImageData: {
        images: {
          fallback: {
            src: "avatar.png",
          },
          sources: [
            {
              type: "image/webp",
              srcSet: "avatar-srcSet.webp",
            },
          ],
        },
      },
    },
  },
}

jest.spyOn(Gatsby, "useStaticQuery").mockImplementation(() => data)

describe("Header", () => {
  beforeEach(() =>
    render(
      <Header
        data-testid="header"
        {...{ title: "Title", fullname: "Fullname", pronouns: "Pronouns" }}
      />
    )
  )

  it("renders correctly", () => {
    expect(screen.getByTestId("header")).toMatchSnapshot()
  })
})
