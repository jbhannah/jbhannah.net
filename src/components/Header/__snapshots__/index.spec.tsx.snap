// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="css-xutqz5"
  data-testid="header"
>
  <h1
    class="css-19x6mp6"
    role="heading"
  >
    <a
      aria-current="page"
      class="css-uuv0t2"
      href="/"
    >
      <picture
        class="css-xik1rc"
      >
        <source
          srcset="avatar-srcSet.webp"
          type="image/webp"
        />
        <img
          alt="Title"
          src="avatar.png"
        />
      </picture>
      Fullname
      <small>
        (
        Pronouns
        )
      </small>
    </a>
  </h1>
  <nav
    class="css-7zh605"
  >
    <ul
      class="css-1cxsv5l"
    >
      <li
        class="css-1cc6n5j"
      >
        <a
          class="css-zdg12z"
          href="https://www.example.com"
          rel="noopener"
          target="_blank"
          title="Test Name on Test Service"
        >
          <test-file-stub
            classname="css-1k2mngo"
          />
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
